packages/block-editor/src/components/inspector-controls-tabs/appearance-tab.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * WordPress dependencies
+ */
+import { hasBlockSupport } from '@wordpress/blocks';
+import {
+	PanelBody,
+	__experimentalUseSlotFills as useSlotFills,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockStyles from '../block-styles';
+import DefaultStylePicker from '../default-style-picker';
+import InspectorControls from '../inspector-controls';
+import InspectorControlsGroups from '../inspector-controls/groups';
+
+const AppearanceTab = ( {
+	blockName,
+	clientId,
+	hasBlockStyles,
+	hasSingleBlockSelection = false,
+} ) => {
+	const { border, color, dimensions, typography } = InspectorControlsGroups;
+	const appearanceFills = [
+		...( useSlotFills( border.Slot.__unstableName ) || [] ),
+		...( useSlotFills( color.Slot.__unstableName ) || [] ),
+		...( useSlotFills( dimensions.Slot.__unstableName ) || [] ),
+		...( useSlotFills( typography.Slot.__unstableName ) || [] ),
+	];
+
+	return (
+		<>
+			{ ! appearanceFills.length && (
+				<span className="block-editor-block-inspector__no-block-tools">
+					{ hasSingleBlockSelection
+						? __( 'This block has no appearance tools.' )
+						: __(
+								'The selected blocks have no appearance tools.'
+						  ) }
```

Comment:
The wording here could likely be improved. I wasn't sure about "appearance tools". Would design or block tools be any better? Any other suggestions?

Reply:
It might be good to linguistically link the functionality here to global styles, IE "This block has no style options.".

Reply:
Thank you, that's much better. (https://github.com/WordPress/gutenberg/pull/45005/commits/01fbbf1eeedeb47d681c899be25650182a5538a1)